libgoal/transactions.go

```diff
@@ -217,19 +298,19 @@ func (c *Client) MakeUnsignedGoOnlineTx(address string, part *account.Participat
 
 	// Choose which participation keys to go online with;
 	// need to do this after filling in the round number.
-	if part == nil {
-		bestPart, err := c.chooseParticipation(parsedAddr, basics.Round(firstValid))
-		if err != nil {
-			return transactions.Transaction{}, err
-		}
-		part = &bestPart
+	part, err := c.chooseParticipation(parsedAddr, basics.Round(firstValid))
+	if err != nil {
+		return transactions.Transaction{}, err
 	}
 
 	parsedFrstValid := basics.Round(firstValid)
 	parsedLastValid := basics.Round(lastValid)
 	parsedFee := basics.MicroAlgos{Raw: fee}
 
-	goOnlineTransaction := part.GenerateRegistrationTransaction(parsedFee, parsedFrstValid, parsedLastValid, leaseBytes, cparams.EnableStateProofKeyregCheck)
```

Comment:
this function still exists, why not re-using it?

Reply:
This one I used from @winder's PR.  They do seem similar but I think there are some additional if statements in `generateRegistrationTransaction`...comments @winder ?

Reply:
The difficulty here is that we're transitioning away from `PersistedParticipation`. In some places we still have that object, but in others it has been replaced with `generated.ParticipationKey`.

It isn't clear since the type isn't visible anywhere, but in this case, the type of `part` has changed so we cannot use `part.GenerateRegistrationTransaction`.